contracts/instance/Instance.sol

```diff
@@ -186,12 +185,11 @@ contract Instance is
         // _instanceAdmin.setTargetFunctionRoleByInstance(targetName, selectors, roleId);
     }
 
-    function setTargetLocked(address target, bool locked)
+    function setLocked(address target, bool locked)
         external 
         onlyOwner()
     {
-        // TODO refactor
-        // _instanceAdmin.setTargetLockedByInstance(target, locked);
+        _componentService.setLockedFromInstance(target, locked);
```

Comment:
What are the reasons to follow Instance -> ComponentService -> InstanceAdmin? 
Proposal: call instance admin directly: Instance -> InstanceAdmin.lockComponentFromIntance()

Reply:
to keep it in line with the concept of only changing state through services.

Reply:
Would argue against because in this case lock by Instance depends from service which is upgradeable.

Reply:
so far we always changed state through service. don't see why we shouldn't do this here. Services are always upgradable.